### PR TITLE
feat(outcomes-consumer): Multithreading

### PR DESCRIPTION
**Pull request by untitaker**
We have performance issues with the outcomes consumer we're running. On top of some obviously slow stuff there's some problematic behavior we can't explain yet. We have very little confidence that this improves the situation we're having, but we're trying it out anyway.

### CODE REVIEW

File: src/sentry/ingest/outcomes_consumer.py [hunk 1]
```diff
@@ -59,56 +62,68 @@ def is_signal_sent(project_id, event_id):
...
-            event_filtered.send_robust(ip=remote_addr, project=project, sender=self.process_message)
-        elif outcome == Outcome.RATE_LIMITED:
-            event_dropped.send_robust(
-                ip=remote_addr, project=project, reason_code=reason, sender=self.process_message
-            )
 
-        # remember that we sent the signal just in case the processor dies before
-        mark_signal_sent(project_id=project_id, event_id=event_id)
+class OutcomesConsumerWorker(AbstractBatchWorker):
+    def __init__(self, multiprocessing, concurrency):
```
Comment (tonyo) [inline src/sentry/ingest/outcomes_consumer.py]: Do we not use `concurrency` anywhere now?

Comment (untitaker) [reply src/sentry/ingest/outcomes_consumer.py]: `concurrency` is not an option for batchingkafkaconsumer if that's what you're asking.

Comment (untitaker) [reply src/sentry/ingest/outcomes_consumer.py]: Ah, you're right. Fixed. (Also tested properly this time)

Comment (tonyo) [reply src/sentry/ingest/outcomes_consumer.py]: 👍

**Review by tonyo — APPROVED** [review]
Just a note, this approach will render `process_message` metric useless, we'll have to look at `process_message.normalized` metric instead.

**Review by RaduW — APPROVED** [review]

**Pull request merged**